wire-ios/Wire-iOS/Sources/UserInterface/DeviceView/DeviceDetailsViewActionsHandler.swift

```diff
@@ -129,25 +133,11 @@ final class DeviceDetailsViewActionsHandler: DeviceDetailsViewActions, Observabl
         guard let certificate = certificate else {
             return
         }
-        let fileName = "e2eiCertifcate.txt"
-        let path = getDocumentsDirectory().appendingPathComponent(fileName)
-        do {
-            try certificate.certificateDetails.write(
-                to: path,
-                atomically: true,
-                encoding: String.Encoding.utf8
-            )
-        } catch {
-            logger.error(error.localizedDescription)
-        }
-    }
-
-    private func getDocumentsDirectory() -> URL {
-        let paths = FileManager.default.urls(
-            for: .documentDirectory,
-            in: .userDomainMask
-        )
-        return paths[0]
+        downloadFileManager.download(
+                    value: certificate.certificateDetails,
+                    fileName: userClient.label ?? "e2ecertificate",
+                    type: "txt"
+                )
```

Comment:
When I see the usage I wonder why this is not async or returning anything. The class name `DownloadFileManager` gives me the assumption this is a general helper and can be used in many places.

Reply:
Yes. Name is a bit misleading. Download here means to save. Let me rename that